kernel/CMakeLists.txt

```
@@ -117,6 +117,35 @@ target_sources_ifdef(CONFIG_OBJ_CORE              kernel PRIVATE obj_core.c)
 
 if(${CONFIG_KERNEL_MEM_POOL})
   target_sources(kernel PRIVATE mempool.c)
+
+  # Import all custom HEAP_MEM_POOL minimum size requirements
+  import_kconfig(CONFIG_HEAP_MEM_POOL_MIN_SIZE_ ${DOTCONFIG} min_size_keys)
+
+  set(max_min_size 0)
+
+  foreach(min_size ${min_size_keys})
+    if(${${min_size}} GREATER ${max_min_size})
```

as a general safety precaution, please quote the variables when expanding.
Empty values / values containing spaces may give very strange and hard to find issues.
```suggestion
  if("${${min_size}}" GREATER "${max_min_size}")
```

---

Done